albatross/core/fit_model.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef ALBATROSS_CORE_FIT_H
+#define ALBATROSS_CORE_FIT_H
+
+namespace albatross {
+
+template <typename ModelType, typename Fit>
+class FitModel {
```

Comment:
This `FitModel` class is the return type of `ModelBase.get_fit_model(dataset)`.  It's basically just a container which keeps a model and a fit produced by that model together.  This is also used to enforce that you can only produce a prediction after having called fit.  Notice that `ModelBase` doesn't have a `get_predictions` method, you have to first get a fit, then make predictions from it.  Ie, `ModelBase.get_fit_model(dataset).get_predictions(features)`.